src/gf.c

```diff
@@ -800,12 +800,15 @@ JL_DLLEXPORT int jl_isa_compileable_sig(
 
     size_t i, np = jl_nparams(type);
     size_t nargs = definition->nargs; // == jl_field_count(jl_unwrap_unionall(decl));
-    if (definition->isva ? np <= nargs : np != nargs)
-        return 0;
+    if (np != nargs) {
+        if (!definition->isva || (np <= nargs && jl_is_vararg_type(jl_tparam(type, np - 1)))) {
```

Comment:
I don't quite understand this condition. Why does the vararg matter?

Reply:
Only Vararg functions may be compiled for Vararg signatures

Reply:
Right, but isn't this asking the opposite?

Reply:
Right, so this disallows anything of the form say `Tuple{typeof(f), Int...}`, for say `f(a::Int, b::Int...)` which may be what you intended, but I thought you may have intended the opposite.

Reply:
Right, this check is for ensure that we have enough parameters to form a Vararg. For np == nargs, we expect to just compile a concrete signature